.changeset/tabaxi-fellas-inspire.md

```diff
@@ -0,0 +1,141 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+"__section": tree
+---
+TableSchema (alpha) APIs have been added
+
+Users may now represent dynamic, tabular data using Shared Tree.
+New APIs have been added under a `TableSchema` namespace for creating column, row, and table node schema.
+
+Note: these APIs require the use of [SchemaFactoryAlpha](https://fluidframework.com/docs/api/fluid-framework/schemafactoryalpha-class).
+
+#### Creating a table with default column and row schema
+
+You can craft a table with defaults via `TableSchema.createTable`, without specifying custom column or row schema.
+Note that you will still be required to provide a schema for the cells that will appear in the table.
+
+```typescript
+class Cell extends schemaFactory.object("TableCell", {
+	value: schemaFactory.string,
+}) {}
+
+class Table extends TableSchema.createTable({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+const table = new Table({
+	columns: [{ id: "column-0" }],
+	rows: [{ id: "row-0", cells: {} }],
+});
+```
+
+#### Creating a table with custom column and row schema
+
+If you need to associate additional data with your rows and columns, you can customize your schema via `TableSchema.createColumn` and `TableSchema.createRow`.
+These schema can then be provided to `TableSchema.createTable`.
+
+```typescript
+class Cell extends schemaFactory.object("TableCell", {
+	value: schemaFactory.string,
+}) {}
+
+class ColumnProps extends schemaFactory.object("TableColumnProps", {
+	// Column label to display.
+	label: schemaFactory.string,
+	// The type of data represented by the cells. Default: string.
```

Comment:
I'm a bit confused here. Is the intended implication here that this optional field stores some string describing some concept which the cells in the column "represent", and the default concept (when this field us undefined is described the string "string" which isn't quoted for some reason?

Also its best practice to document fields with doc comments so IntelliSense works for them.

Regardless, I think the same argument I made about not defining a Cell schema as is districts from the example applies here as well. I think some simpler inline schema in the createColumn would be more clear.

Reply:
The example has been simplified.